### PR TITLE
UI-428 Checkbox - Lightning Component

### PR DESCRIPTION
**Pull request by mallik-svmx**
- Refactored Checkbox and CheckboxGroup components, designed to work independently
- Removed ref attribute, making it controlled input
- Removed rest props for clarity and to have those props listed on documentation
- Added basic unit tests
- Have retained the ability to create columns while creating CheckboxGroup components
- Addressed UI-605 Lightning needs custom css added to storybook & exported with package

### CODE REVIEW

File: lib/components/Checkbox/Checkbox.js [hunk 1]
```diff
@@ -1,63 +1,97 @@
...
+      <label htmlFor={id} className={classes}>
+        <input
+          checked={isChecked}
+          disabled={disabled}
+          id={id}
+          onChange={this.handleChange}
+          type="checkbox"
+          value={value}
+        />
+        <span className="slds-checkbox--faux" style={{ marginRight: '.5rem' }} />
```
Comment (mallik-svmx) [inline lib/components/Checkbox/Checkbox.js]: pls ignore this inline style for now, will have it moved over to a different file once sass loader is configured.

**Comment by mallik-svmx** [comment]
@joshsylvester Addressed review comments and PR also include UI-605: sass-loader

File: lib/components/Checkbox/CheckboxGroup.js [hunk 2]
```diff
@@ -1,107 +1,122 @@
...
-          className="slds-form-element__control"
-        >
-          { React.Children.map(children, this.renderControl) }
-          {
-            errorMessage ?
-              <div className="slds-form-element__help">{ errorMessage }</div> :
-              undefined
-          }
+        <div className="slds-form-element__control">
+          {React.Children.map(children, this.renderControl)}
```
Comment (joshsylvester) [inline lib/components/Checkbox/CheckboxGroup.js]: minor nitpick - can we either rename this to `renderCheckbox`, or just put the `React.cloneElement` inline right here? it doesn't seem to be referenced anywhere else.

Comment (mallik-svmx) [reply lib/components/Checkbox/CheckboxGroup.js]: Yep. Just renamed it so, makes it easier for any future enhancements.

**Pull request merged**